### PR TITLE
♻️🔨  ``api-server``: fixes mypy issues and add type-check step in CI

### LINKED ISSUE #3551: Flaky tests/unit/with_dbs/02/test_project_db.py::test_check_project_permissions[read|write|delete]
## [int] web-server 02

- [ ] ``tests/unit/with_dbs/02/test_project_db.py::test_check_project_permissions[read|write|delete]``. 
    - hangs and timesout somewhere after the test above
    -  Example [here](https://github.com/ITISFoundation/osparc-simcore/actions/runs/3197862411/jobs/5221651237)

### PR DESCRIPTION
**Pull request by pcrespov**
## What do these changes do?

Fixes mypy errors in ``api-server``  and adds ``type-check`` step in CI

## Related issue/s

- related #4071
- fixes #3551: flaky

## How to test

```cmd
cd services/api-server
make install-dev
make mypy
```
should not show any error

### CODE REVIEW

**Review by matusdrobuliak66 — APPROVED** [review]

File: services/api-server/src/simcore_service_api_server/api/routes/users.py [hunk 1]
```diff
@@ -22,11 +23,12 @@ async def get_my_profile(
...
     # FIXME: temporary patch until web-API is reviewed
-    data["role"] = data["role"].upper()
+
     try:
+        data["role"] = data["role"].upper()  # type: ignore
         profile = Profile.parse_obj(data)
-    except ValidationError as err:
-        logger.exception("webserver invalid response")
+    except (ValidationError, KeyError) as err:
+        logger.error("webserver invalid response %s", data, exc_info=True)
```
Comment (GitHK) [inline services/api-server/src/simcore_service_api_server/api/routes/users.py:31]: Maybe rethink theology message a well. What does `webserver invalid response` mean to someone looking in the logs?
The reason seems to be related to invalid request data received.

Comment (GitHK) [reply services/api-server/src/simcore_service_api_server/api/routes/users.py:31]: Also the role.upper() could be handled by a validator in the Profile model at this point.

**Review by GitHK — APPROVED** [review]

File: services/api-server/src/simcore_service_api_server/api/routes/users.py (hunk 1 above)
Comment (pcrespov) [reply services/api-server/src/simcore_service_api_server/api/routes/users.py:31]: @GitHK  I agree. The whole error handling in the api-server is very very poor :-(  
I focused here in mypy but will review all error handling and logging in the api-server

**Review by sanderegg — APPROVED** [review]
Just if I may, there are 5 code issues found by code climate. did you see that?

**Comment by pcrespov** [comment]
> Just if I may, there are 5 code issues found by code climate. did you see that?

I will take care of them

**Pull request merged**